Sources/AnimatedAssignSubscriber.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  AnimatedAssignSubscriber.swift
+//
+//  Created by Marin Todorov on 5/3/20.
+//
+
+import Foundation
+import Combine
+
+#if canImport(UIKit)
+import UIKit
+
+/// A list of animations that can be used with `Publisher.assign(to:on:animation:)`
+public enum AssignTransition {
+    public enum Direction {
+        case top, bottom, left, right
+    }
+    
+    /// Flip from either bottom, top, left, or right.
+    case flip(direction: Direction, duration: TimeInterval)
+    
+    /// Cross fade with previous value.
+    case fade(duration: TimeInterval)
+    
+    /// A custom animation. Do not include your own code to update the target of the assign subscriber.
+    case animation(duration: TimeInterval, options: UIView.AnimationOptions, animations: () -> Void, completion: ((Bool) -> Void)?)
+}
+
+extension Publisher where Self.Failure == Never {
+	/// Behaves identically to `Publisher.assign(to:on:)` except that it allows the user to
+	/// "wrap" emitting output in an animation transition.
+	///
+	/// For example if you assign values to a `UILabel` on screen you
+	/// can make it flip over when each new value is set:
+	///
+	/// ```
+	/// myPublisher
+	///   .assign(to: \.text,
+	///             on: myLabel,
+	///             animation: .flip(direction: .bottom, duration: 0.33))
+	/// ```
+	///
+	/// You may also provide a custom animation block, as follows:
+	///
+	/// ```
+	/// myPublisher
+	///   .assign(to: \.text, on: myLabel, animation: .animation(duration: 0.33, options: .curveEaseIn, animations: { _ in
+	///     myLabel.center.x += 10.0
+	///   }, completion: nil))
+	/// ```
+    public func assign<Root: UIView>(to keyPath: ReferenceWritableKeyPath<Root, Self.Output>, on object: Root, animation: AssignTransition) -> AnyCancellable {
```

Comment:
Not sure why this looks misaligned... tabs and spaces together? 
![image](https://user-images.githubusercontent.com/605076/81464305-141b2a00-91c9-11ea-8e09-fbe1f1b9da43.png)